R/oi_im_pavement_width.R

```diff
@@ -0,0 +1,41 @@
+#' Function to assess pedestrian infrastructure widths.
+#' 
+#' @usage oi_im_pavement_width(osm_sf)
+#' @param osm_sf - A `sf` and `data.frame` object containing OpenStreetMap 
+#'   infrastructure data, obtained from the 
+#'   [`osmextract`](https://github.com/ropensci/osmextract) package.
+#' @return the `osm_sf` is returned with additional columns `openinfra_im_width`
+#'   and `openinfra_im_width_est` with recategorised way width and estimated 
+#'   width value ranges appended respectively.
+#' @details Note: the `osm_sf` must contain the following tags: `c("width", 
+#'   "est_width")`
+#' @export 
+#' @examples
+#' data = example_data
+#' output = oi_im_pavement_width(data)
+#' plot(output["openinfra_im_width_est"])
+#' # Uncomment below to plot other columns
+#' # plot(output["openinfra_im_width"])
+
+oi_im_pavement_width = function(osm_sf){
+  # Assesses way width - either < 1.5 meters, 1.5-2 meters, or > 2 meters.
+  osm_sf_im = osm_sf %>% dplyr::mutate(openinfra_im_width = width %>% 
+    readr::parse_number(),
```

Comment:
Great use of parse number. How does it hand units in ft?

Reply:
I will try and add a check for this. Do we convert to metres or allow ft. as an output too? (vote convert to metres so not too many units in the column, though more can be merrier!)

Reply:
I think OSM docs say that all widths should be in metres. Worth doing a bit of sanity checking though if poss. For now, it's fine, and those checks can be a separate issue/bit of code.

Reply:
Had a look at width values and it seems a lot lack units. 

My initial thought (and one I am rolling with for now): 
apply `stringr::str_detect()` to look for ("ft", "f", "feet") to identify widths in feet -> apply parse_number() & multiply by 0.3048 to convert to metres (1m = 0.3048ft). 

Could also just leave as ft, but not want to create extra work for people who may use this data.

If a values is purely numeric, assume metres? (could add checks to _assume_ units i.e. ' " ' may indicate a foot/inch measurement (6"8), but this could get messy)